### PR TITLE
CNF-4018 Enable Flexible Registration for Multiple PTP Events clock-class-change,status and os-sync

### PR DESCRIPTION
**Pull request by aneeshkp**
Enable Flexible Registration for Multiple PTP Events
This adds the ability to subscribe to multiple events.
For each Publisher/Subscriber you create separate AMQ sender/listener objects
Signed-off-by: Aneesh Puttur <aneeshputtur@gmail.com>

### CODE REVIEW

File: examples/consumer/main.go [hunk 1]
```diff
@@ -145,25 +136,32 @@ RETRY:
...
-		if status, b := rc.PostWithReturn(subURL, b); status == http.StatusCreated {
-			return b
+		if status, subB = rc.PostWithReturn(subURL, subB); status != http.StatusCreated {
+			err = fmt.Errorf("subscription creation api at %s, returned status %d", subURL, status)
+			return
 		}
-		log.Errorf("subscription create returned error %s", string(b))
 	} else {
-		log.Errorf("failed to create subscription ")
+		log.Error("failed to marshal subscription ")
```
Comment (jzding) [inline examples/consumer/main.go:158]: add return here.

Comment (aneeshkp) [reply examples/consumer/main.go]: done

**Comment by jzding** [comment]
look good to me. I will let Nishant to add the label so that he has a chance to review.

File: plugins/mock/mock_plugin.go [hunk 2]
```diff
@@ -113,19 +113,19 @@ func createMockEvent(pub pubsub.PubSub) (ceEvent.Event, error) {
 	data := ceEvent.Data{
 		Version: "v1",
 		Values: []ceEvent.DataValue{{
-			Resource:  string(ptp.SyncStatusState),
+			Resource:  "/mock",
 			DataType:  ceEvent.NOTIFICATION,
 			ValueType: ceEvent.ENUMERATION,
 			Value:     ptp.ACQUIRING_SYNC,
```
Comment (nishant-parekh) [inline plugins/mock/mock_plugin.go:119]: nit: may be want remove any reference to ptp-event in mock_plugin. can we set this to an arbitary string

File: hack/deploy_test.sh [hunk 3]
```diff
@@ -24,17 +25,27 @@ if [ "$action" == "undeploy" ]; then
...
  deploy_consumer $action $NamespaceConsumerTesting $NamespaceAMQTesting || true
  deploy_producer $action $NamespaceProducerTesting $NamespaceAMQTesting || true
-
-fi
-
-
-
-
-
+fi
```
Comment (nishant-parekh) [inline hack/deploy_test.sh:51]: add an extra carriage return to avoid github warnings

File: plugins/ptp_operator/ptp_operator_plugin.go [hunk 4]
```diff
@@ -70,17 +70,25 @@ func Start(wg *sync.WaitGroup, configuration *common.SCConfiguration, fn func(e
 	config = configuration
 	// register metrics type
 	ptpMetrics.RegisterMetrics(nodeName)
+	publishers = InitPubSubTypes()
 
 	// 1. Create event Publication
-	var pub pubsub.PubSub
+	// 1. Create event Publication
```
Comment (nishant-parekh) [inline plugins/ptp_operator/ptp_operator_plugin.go:76]: remove

File: plugins/ptp_operator/ptp_operator_plugin.go [hunk 5]
```diff
@@ -314,3 +325,21 @@ func HasEqualInterface(a []*string, b []*ptp4lconf.PTPInterface) bool {
...
+		Resource:  ptp.OsClockSyncState,
+	}
+	publishers[ptp.PtpClockClassChange] = &ptpTypes.EventPublisherType{
+		EventType: ptp.PtpClockClassChange,
+		Resource:  ptp.PtpClockClass,
+	}
+	publishers[ptp.PtpStateChange] = &ptpTypes.EventPublisherType{
+		EventType: ptp.PtpStateChange,
+		Resource:  ptp.PtpLockState,
+	}
```
Comment (nishant-parekh) [inline plugins/ptp_operator/ptp_operator_plugin.go:343]: There are a few other event Types defined in the ptp event types file do we need to add the publishers for those eventTypes too?
https://github.com/redhat-cne/sdk-go/blob/main/pkg/event/ptp/types.go

File: plugins/ptp_operator/ptp_operator_plugin.go [hunk 6]
```diff
@@ -256,7 +264,10 @@ func Start(wg *sync.WaitGroup, configuration *common.SCConfiguration, fn func(e
 		return nil
 	}
 	log.Infof("setting up status listener")
```
Comment (nishant-parekh) [inline plugins/ptp_operator/ptp_operator_plugin.go:266]: update log message to: `setting up status listener for each publisher type`

**Review by nishant-parekh — CHANGES_REQUESTED** [review]
overall looks good some small comments

File: plugins/mock/mock_plugin.go (hunk 2 above)
Comment (aneeshkp) [reply plugins/mock/mock_plugin.go:119]: fixed

File: plugins/ptp_operator/ptp_operator_plugin.go (hunk 5 above)
Comment (aneeshkp) [reply plugins/ptp_operator/ptp_operator_plugin.go:343]: Not yet, right now we only support 3 types of events that are implemented. until we don't know what is needed, we are not adding it. The list is from API spec document

File: plugins/ptp_operator/ptp_operator_plugin.go (hunk 6 above)
Comment (aneeshkp) [reply plugins/ptp_operator/ptp_operator_plugin.go:266]: fixed

**Review by nishant-parekh — APPROVED** [review]

**Pull request merged**